src/Client/Profile/ProfileClient.php

```diff
@@ -151,7 +151,7 @@ public function getBounces(int $id, array $poolAttributeList, int $timestampStar
      * @return HashMapInterface
      * @throws EmptyResultException
      */
-    public function getById(int $id): HashMapInterface
+    public function getById(string $id): HashMapInterface
```

Comment:
I'd say, this breaks BC, doesn't it? Maybe we need to drop the typehint until the next major version and add a union typehint `int|strig`.

Reply:
It's a BC, it was just a quick Shot 
. In fact, I would tend to establish a completely new method getBySid(String $sid). Seems to me to be the best solution in general

Reply:
> . In fact, I would tend to establish a completely new method getBySid(String $sid). Seems to me to be the best solution in general

I second that.